feeds/mediatek-sdk/mediatek/mt7981/base-files/lib/upgrade/platform.sh

```diff
@@ -32,6 +32,9 @@ platform_do_upgrade() {
 		fi
 		nand_do_upgrade "$1"
 		;;
+	sonicfi,rap630w-211g)
+		/tmp/nand_sonicfi_rap630w_211g.sh "$1"
```

Comment:
what is this needed for ?

Reply:
The main reason is that when upgrading the image, we cannot use the original nand.sh because we are currently using a private structured image. Therefore, we need to write a separate private upgrade script.

Reply:
The main reason is that when upgrading the image, we cannot use the original nand.sh because we are currently using a private structured image. Therefore, we need to write a separate private upgrade script.

Reply:
can we use a default image inside TIP images ?

Reply:
No, we need to use a third-party boot code during the production process to support switching between dual images.

Reply:
please provide details of why this is needed

Reply:
Rap630w_211G is using a third-party boot code, so its image structure needs to be compatible with the boot code. Therefore, it needs to make modifications to the sysupgrade process in order to work properly. Therefore, it cannot use the nand.sh function that comes with tip. At the same time, it needs to take care not to affect other tip model projects. Therefore, it has separated the private nand.sh and used the private script during the sysupgrade process.

Reply:
@blogic   Hello, may I ask if our push request has been approved this time? Thank you!

Reply:
/tmp/nand_sonicfi_rap630w_211g.sh is not part of the patch and the resulting image would not be upgradable

Reply:
also there is no valid Signed-off-by line in the commit

Reply:
@blogic :
In the file (patches/0072-add-sonicfi-rap630w-211g.patch),the patch will create the file (package/base-files/files/lib/upgrade/sonicfi/nand_sonicfi_rap630w_211g.sh) and add some codes for the 'package/base-files/files/sbin/sysupgrade'.

+board=$(board_name)
+case "$board" in
+	sonicfi,rap630w-211g)
+	cp /lib/upgrade/sonicfi/nand_sonicfi_rap630w_211g.sh /tmp/nand_sonicfi_rap630w_211g.sh
+	chmod 777 /tmp/nand_sonicfi_rap630w_211g.sh
+	;;
+esac
+

When the 'sonicfi rap630w-211g' Router is upgrading the image.Sysupgrade script will copy the 'nand_sonicfi_rap630w_211g.sh' from the '/lib/upgrade/sonicfi' directory to the '/tmp/' directory.

Reply:
@CybertanYingWei why does it need to be copied to tmp ? the file should also be located in feeds/mediatek/mediatek/base-files/....

Reply:
Because when performing the step of writing NAND flash, the system has already umounted all partitions, and the path (/lib/upgrade/) is no longer accessible. Therefore, it is necessary to copy it to '/tmp/' in advance in order to access the files inside normally.

Reply:
@CybertanYingWei openwrt has a facility for this. you can set
RAMFS_COPY_BIN=''
RAMFS_COPY_DATA=''
inside platform.sh

Reply:
Okay, I will modify the code again, thank you!

Reply:
I have revised the relevant code as per your request. Now, please close this push request and push the new code again. Thank you!